runtime/src/main/java/org/corfudb/protocols/wireprotocol/orchestrator/AbstractWorkflow.java

```diff
@@ -12,24 +12,33 @@
  * Created by Maithem on 10/25/17.
  */
 
-public interface Workflow {
+public abstract class AbstractWorkflow {
 
     /**
      * Gets the unique identifier of this workflow instance
      * @return id of a particular instance
      */
-    UUID getId();
+    public abstract UUID getId();
 
     /**
      * Return the name of this workflow
      * @return workflow's name
      */
-    String getName();
+    public abstract String getName();
 
     /**
      * Returns the ordered that are associated with
      * this workflow.
      * @return List of actions
      */
-    List<Action> getActions();
+    public abstract List<Action> getActions();
+
+    /**
+     * Returns whether all the actions completed successfully or not.
+     * @return true if all actions completed successfully
+     */
+    public boolean completed() {
+        return getActions().stream()
+                .anyMatch(a -> a.equals(ActionStatus.COMPLETED));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this call to "equals"; comparisons between unrelated types always return false. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2159)

Reply:
That's a bug. Actions != ActionStatus

Reply:
Done.